File: deprecated/hybrid-agent-pattern/langgraph_python_example/hybrid_agent.py
import os
import requests
import re

def symbolic_module(user_input):
    match = re.match(r'add (\d+) and (\d+)', user_input.lower())
    if match:
        sum_result = int(match.group(1)) + int(match.group(2))
        return f'Symbolic: The sum is {sum_result}.'
    return None

def ml_module(user_input):
    api_key = os.getenv('MISTRAL_API_KEY')
    if not api_key:
        raise ValueError('Please set MISTRAL_API_KEY in your .env file')
    url = 'https://api.mistral.ai/v1/chat/completions'
    headers = {
        'Authorization': f'Bearer {api_key}',
        'Content-Type': 'application/json',
    }
    data = {
        'model': 'mistral-tiny',
        'messages': [
            {'role': 'system', 'content': 'You are an ML module. Answer the user query as best as you can.'},
            {'role': 'user', 'content': user_input},
        ],
    }
    response = requests.post(url, headers=headers, json=data)
    response.raise_for_status()
    return 'ML: ' + response.json()['choices'][0]['message']['content'].strip()

def main():
    user_input = input('Ask the hybrid agent a question: ')
    symbolic_result = symbolic_module(user_input)
    if symbolic_result:
        final_result = symbolic_result
    else:
        final_result = ml_module(user_input)
    print('\nHybrid Agent Response:\n', final_result)

if __name__ == '__main__':
    main()

File: deprecated/hybrid-agent-pattern/autogen_typescript_example/hybrid_agent.ts
import dotenv from 'dotenv';
dotenv.config();
import axios from 'axios';
import readline from 'readline';

const MISTRAL_API_KEY = process.env.MISTRAL_API_KEY;
const MISTRAL_API_URL = 'https://api.mistral.ai/v1/chat/completions';

if (!MISTRAL_API_KEY) {
  console.error('Please set MISTRAL_API_KEY in your .env file');
  process.exit(1);
}

// Simple symbolic rule-based module
function symbolicModule(input: string): string | null {
  if (input.toLowerCase().includes('add')) {
    const match = input.match(/add (\d+) and (\d+)/i);
    if (match) {
      const sum = parseInt(match[1]) + parseInt(match[2]);
      return `Symbolic: The sum is ${sum}.`;
    }
  }
  return null;
}

async function mlModule(input: string): Promise<string> {
  const messages = [
    { role: 'system', content: 'You are an ML module. Answer the user query as best as you can.' },
    { role: 'user', content: input },
  ];
  const response = await axios.post(
    MISTRAL_API_URL,
    {
      model: 'mistral-tiny',
      messages,
    },
    {
      headers: {
        'Authorization': `Bearer ${MISTRAL_API_KEY}`,
        'Content-Type': 'application/json',
      },
    }
  );
  return 'ML: ' + response.data.choices[0].message.content.trim();
}

async function main() {
  const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
  });

  rl.question('Ask the hybrid agent a question: ', async (userInput) => {
    // Try symbolic module first
    const symbolicResult = symbolicModule(userInput);
    let mlResult = '';
    if (!symbolicResult) {
      mlResult = await mlModule(userInput);
    }
    // Integrate results
    let finalResult = symbolicResult || mlResult;
    console.log('\nHybrid Agent Response:\n', finalResult);
    rl.close();
  });
}

main();
